### PR TITLE
add join timeout and make it configurable

### PR DESCRIPTION
**Pull request by smitsohu**
per this patch, firejail tries for 5 seconds to join a sandbox, and only then exits with an error. I'm not sure if the defaults are chosen wisely, you may want to adjust both the default timeout and the frequency of the checks (every 100 msec currently)

see #2139

### CODE REVIEW

File: src/firejail/join.c [hunk 1]
```diff
@@ -255,32 +255,114 @@ static void extract_umask(pid_t pid) {
 	fclose(fp);
 }
 
+// return 0 if the sandbox identified by pid is not fully set up yet or if
```
Comment (topimiettinen) [inline src/firejail/join.c]: Could be nicer with 'bool'.

**Comment by topimiettinen** [comment]
Looks good.

**Comment by smitsohu** [comment]
@topimiettinen I will update is_ready_for_join() so it returns a boolean, thanks for the review

**Comment by netblue30** [comment]
All fine with me, so check it in when you are done.

**Pull request merged**